File: 1.py
class Solution:
    # @return a tuple, (index1, index2)
    def twoSum(self, num, target):
    	dic = {}
    	for i in num:
    		dic[i] = 1
    	for i in range(len(num)):
    		if (target - num[i]) in dic:
    			a = target - num[i]
    			for j in range(len(num)):
    				if num[j] == a:
    					break
    			if i == j:
    				continue
    					
    			if i > j:
    				return (j+1, i+1)
    			else:
    				return (i+1, j+1)

a = Solution()
(i, j) = a.twoSum([3,2,4], 6)
print "index1=%d, index2=%d" % (i, j)

File: 1.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  dct = {}
  for num in nums
    dct[num] = 1
  end
  for num in nums
    if dct.has_key?(target-num)
      a = [num, target-num]
      return a.min, a.max
    end
  end
end

p two_sum([2, 7, 11, 15], 9)
